src/worker.cpp

```diff
@@ -651,12 +652,65 @@ bool Worker::isCPUIOGoodOnly(const std::string& pgKeyword)
     return false;
 }
 
-void Worker::primeInventory(const types::IPZVpdMap& ipzVpdMap,
-                            types::ObjectMap primeObjects)
+bool Worker::primeInventory(const std::string& i_vpdFilePath)
 {
-    // TODO implementation
-    (void)ipzVpdMap;
-    (void)primeObjects;
+    logging::logMessage("primeInventory called for FRU " + i_vpdFilePath);
+
+    if (i_vpdFilePath.empty())
+    {
+        logging::logMessage("Empty VPD file path given");
+        return false;
+    }
+
+    if (m_parsedJson.empty())
+    {
+        logging::logMessage("Empty JSON detected for " + i_vpdFilePath);
+        return false;
+    }
+
+    types::ObjectMap l_objectInterfaceMap;
+    for (const auto& l_Fru : m_parsedJson["frus"][i_vpdFilePath])
+    {
+        types::InterfaceMap l_interfaces;
+        sdbusplus::message::object_path l_fruObjectPath(l_Fru["inventoryPath"]);
+
+        types::VPDMapVariant l_emptyValue;
```

Comment:
Why is this required? Why don't you use empty value directly at calling point?

Reply:
populateInterfaces API is expecting reference variable, so can't send rvalue to it.

Reply:
So? You are saying that will a compile time error?

Reply:
yes, will get compile time error.

Reply:
Again, you know/checked or you are assuming?

Reply:
I try passing variable of types::VPDMapVariant  type directly to populateInterfaces(), got an error yesterday.
and I even tried with small test code today, got the error. 
Please find the error snippet,
test_ref.cpp:9:14: error: cannot bind non-const lvalue reference of type ‘int&’ to an rvalue of type ‘int’
    9 |         func(3);
      |              ^
test_ref.cpp:5:16: note:   initializing argument 1 of ‘void func(int&)’
    5 | void func(int &a){
      |           ~~~~~^

Reply:
So, the test case that you have put here and the actual scenario is same?

Reply:
yes, the above test scenario is missing const declaration, with const we can pass the rvalue.
updated the code, passing monostate directly now .